src/query/Index.cpp

```diff
@@ -287,32 +285,45 @@ ProjectionMatch Index::findCellOrProjection(const Eigen::VectorXd &location, int
 
 ProjectionMatch Index::findVertexProjection(const Eigen::VectorXd &location)
 {
-  auto match = getClosestVertex(location);
-  return {mapping::Polation{_mesh->vertices()[match.index]}, match.distance};
+  auto              match = getClosestVertex(location);
+  mapping::Polation polation{location, _mesh->vertices()[match.index]};
+  return {polation, polation.distance()};
 }
 
 ProjectionMatch Index::findEdgeProjection(const Eigen::VectorXd &location, int n)
 {
-  auto matchedEdges = getClosestEdges(location, n);
-  for (const auto &match : matchedEdges) {
+  std::vector<ProjectionMatch> candidates;
+  candidates.reserve(n);
+  for (const auto &match : getClosestEdges(location, n)) {
     auto polation = mapping::Polation(location, _mesh->edges()[match.index]);
     if (polation.isInterpolation()) {
-      return {polation, match.distance};
+      candidates.push_back({polation, polation.distance()});
     }
   }
```

Comment:
Looks a bit like `copy_if` could be used, not completely sure.

Reply:
It is a mix between a transform `Match -> (Polation, distance)` and a filter operation `(Polation, distance) -> bool`.
We could implement this with boost ranges, but as so often in C++, it's more readable to use loops like this.